### ISSUE TITLE
Confusion around configuration setting newlib-obsolete-math

### ISSUE THREAD

**Issue by rdiez** [picolibc/picolibc#282]
I am confused about configuration settting `newlib-obsolete-math`.

The documentation in doc/build.md states that its default value is `true`, but the code in meson.build is:

```
if newlib_obsolete_math == 'auto'
  obsolete_math_value = false
elif newlib_obsolete_math == 'true'
  obsolete_math_value = 1
elif newlib_obsolete_math == 'false'
  obsolete_math_value = 0
endif
```
I wouldn't expect a mixture of `0`, `1` and `false` as valid values for a variable, but more importantly, the default seems to be `false`, and not `true`.

The documentation in meson_options.txt states a different default though:

```
option('newlib-obsolete-math', type: 'combo', choices: ['true', 'false', 'auto'],
       value: 'auto',
       description: 'Use float type for float valued math routines (default: automatic based on platform)')
```
And the description is not about "use old code", but something about the "float" type.

**Comment by keith-packard** [comment]
Yeah, there is a potential for confusion here. A 'false' value for meson leaves the target CPP symbol undefined, any other value is used as the CPP symbol value. In this case, leaving the symbol undefined makes the library use its default value rather than using the value specified in the configuration.

And picolibc has changed what the default value is over time -- originally, I used the same default as newlib -- which meant that the 'new' math functions were always used. Then I discovered that those new functions used double computations even for float-valued operations, which really sucked on M4 cores with a 32-bit FPU. So I changed the default to use the old code on targets without a 64-bit FPU. *Then* I discovered that the new code was less accurate than the old code in many areas, so I gave up and switched the default to the old code in all cases.

There are probably places in the docs needing clarification about this ...
